src/phpDocumentor/ConfigurationFactory.php

```diff
@@ -0,0 +1,354 @@
+<?php
+
+namespace phpDocumentor;
+
+final class ConfigurationFactory
+{
+    /**
+     * @var Uri
+     */
+    private $uri;
+
+    /**
+     * @var string
+     */
+    private $schemaPath;
+
+    /**
+     * @param Uri    $uri
+     * @param string $schemaPath
+     */
+    public function __construct(Uri $uri, $schemaPath = '../../data/xsd/phpdoc.xsd')
+    {
+        $this->validate($uri);
+
+        $this->uri        = $uri;
+        $this->schemaPath = $schemaPath;
+    }
+
+    /**
+     * Replace the location of the configuration file.
+     *
+     * @param Uri $uri
+     *
+     * @return ConfigurationFactory
+     */
+    public function replaceLocation(Uri $uri)
+    {
+        return static::__construct($uri);
```

Comment:
please set the uri property here instead of calling the constructor. If someone were to use a different $schemaPath then it would revert to the default after calling this method